src/NanoMsg.cpp

```diff
@@ -0,0 +1,49 @@
+#include "NanoMsg.h"
+#include <cstring>
+#include <stdexcept>
+#include <g3log/g3log.hpp>
+/**
+ * Constructor for copying a packet hash vector into a nanomsg buffer
+ */
+NanoMsg::NanoMsg(PacketHashVector& data) : sent(false) {
+   auto data_size = data.size() * sizeof(std::pair<void*, unsigned int>);
+   buffer = nn_allocmsg(data_size, 0);
```

Comment:
2) @yannick-polius, @john-gress: We can discuss naming since we have moved this to github.com
     if we decide to follow the same 'naming policy' as for our other code, and the rest of the code at Queuenado then you need to change the names of the class members

```
 void* buffer;
 bool sent;
```
to 
```
 void* mBuffer;
 bool mSent;
```